py3status/modules/velib_metropole.py

```diff
@@ -208,46 +203,71 @@ def _manipulate(self, data):
             )
 
             # flat, camel to snake and cast...
-            for k, v in self.py3.flatten_dict(station, delimiter="_").items():
-                new_station[self._camel_to_snake_case(k)] = self._cast_number(v)
+            for key, value in self.py3.flatten_dict(station, delimiter="_").items():
+                key = self._camel_to_snake_case(key)
+                if key in self.placeholders:
+                    new_station[key] = self._cast_number(value)
 
             # station_due_date_s: station due date in dateime iso format
+            if all(
+                "station_due_date" in list for list in [new_station, self.placeholders]
+            ):
+                station_due_date = datetime.fromtimestamp(
+                    new_station["station_due_date"]
+                )
+                new_station.update({"station_due_date_s": station_due_date.isoformat()})
+
             # station_state_code: station code for thresholds
-            station_due_date = datetime.fromtimestamp(
-                int(new_station["station_due_date"])
-            )
-            new_station.update(
-                {
-                    "station_due_date_s": station_due_date.isoformat(),
-                    "station_state_code": int(
-                        self.station_states[new_station["station_state"]]
-                    ),
-                }
-            )
+            if all(
+                "station_state" in list for list in [new_station, self.placeholders]
+            ):
+                new_station.update(
+                    {
+                        "station_state_code": int(
+                            self.station_states[new_station["station_state"]]
+                        )
+                    }
+                )
+
+            stations.append(new_station)
 
-            stations[index] = new_station
-            index += 1
+        # forge return
+        for index, station in enumerate(stations, 1):
+            new_data[index] = station
 
-        return len(stations), stations
+        return new_data
 
     def velib_metropole(self):
-        if self.refresh is True:
-            # get number of stations and station list
-            json = self._get_velib_data()
-            if json:
-                self.number_of_stations, self.stations = self._manipulate(json)
+        # refresh
+        current_time = time()
+        refresh = current_time >= self.idle_time
+
+        # time
+        if refresh:
+            self.idle_time = current_time + self.cache_timeout
+            cached_until = self.cache_timeout
+        else:
+            cached_until = self.idle_time - current_time
 
-        # reset refresh
-        self.refresh = True
+        if not refresh or self.first_start:
+            self.velib_metropole_data = self._get_velib_data()
```

Comment:
Good try. This is refreshing on every scroll now. You're missing `self.toggled` code from `whatismyip` too... to prevent data refreshes on scrolling.